packages/styletron-engine-atomic/src/inject-style-prefixed.js

```diff
@@ -39,26 +45,32 @@ export default function injectStylePrefixed(
       } else {
         // cache miss
         let block = "";
-        const prefixed = prefixAll({[originalKey]: originalVal});
-        for (const prefixedKey in prefixed) {
-          const prefixedVal = prefixed[prefixedKey];
-          const prefixedValType = typeof prefixedVal;
-          if (prefixedValType === "string" || prefixedValType === "number") {
-            const prefixedPair = `${hyphenate(prefixedKey)}:${prefixedVal}`;
-            if (prefixedPair !== propValPair) {
-              block += `${prefixedPair};`;
-            }
-          } else if (Array.isArray(prefixedVal)) {
-            const hyphenated = hyphenate(prefixedKey);
-            for (let i = 0; i < prefixedVal.length; i++) {
-              const prefixedPair = `${hyphenated}:${prefixedVal[i]}`;
+        const addPrefixToBlock = value => {
+          const prefixed = prefixAll({[originalKey]: value});
+          for (const prefixedKey in prefixed) {
+            const prefixedVal = prefixed[prefixedKey];
+            const prefixedValType = typeof prefixedVal;
+            if (prefixedValType === "string" || prefixedValType === "number") {
+              const prefixedPair = `${hyphenate(prefixedKey)}:${prefixedVal}`;
               if (prefixedPair !== propValPair) {
                 block += `${prefixedPair};`;
               }
+            } else if (Array.isArray(prefixedVal)) {
+              const hyphenated = hyphenate(prefixedKey);
+              for (let i = 0; i < prefixedVal.length; i++) {
+                const prefixedPair = `${hyphenated}:${prefixedVal[i]}`;
+                if (prefixedPair !== propValPair) {
+                  block += `${prefixedPair};`;
+                }
+              }
             }
           }
-        }
-        block += propValPair; // ensure original prop/val is last (for hydration)
+        };
+        if (hasFallbacks) {
+          const fallbacks: any = originalVal;
+          for (const fallback of fallbacks) addPrefixToBlock(fallback);
+        } else addPrefixToBlock(originalVal);
+        if (!hasFallbacks) block += propValPair;
```

Comment:
based on how the code is currently written, this can probably go into the `else` block on L72. 

i'm concerned about the lost comment on L61. is `ensure original prop/val is last (for hydration)` relevant to fallback values? Should it be maintained if a fallback is provided?